plugins/chat/app/services/chat/search_chatable.rb

```diff
@@ -45,57 +47,29 @@ def fetch_memberships(guardian:)
     def fetch_users(guardian:, contract:)
       return unless contract.include_users
       return unless guardian.can_create_direct_message?
-      search_users(context, guardian, contract)
+      search_users(context, guardian)
     end
 
     def fetch_groups(guardian:, contract:)
       return unless contract.include_groups
       return unless guardian.can_create_direct_message?
-      search_groups(context, guardian, contract)
+      search_groups(context, guardian)
     end
 
     def fetch_category_channels(guardian:, contract:)
       return unless contract.include_category_channels
-      return if !SiteSetting.enable_public_channels
-
-      ::Chat::ChannelFetcher.secured_public_channel_search(
-        guardian,
-        filter_on_category_name: false,
-        match_filter_on_starts_with: false,
-        filter: context.term,
-        status: :open,
-        limit: 10,
-      )
+      return unless SiteSetting.enable_public_channels
+      search_category_channels(context, guardian)
```

Comment:
Extracted the query into a `search_` method to match other functions in this file.